test/tstCompileOnlyVersionMacros.cpp

```diff
@@ -0,0 +1,24 @@
+#include <ArborX_Config.hpp>
+
+#ifndef ARBORX_VERSION
+static_assert(false, "ARBORX_VERSION macro is not defined!");
+#endif
+
+#ifndef ARBORX_VERSION_MAJOR
+static_assert(false, "ARBORX_VERSION_MAJOR macro is not defined!");
+#endif
+
+#ifndef ARBORX_VERSION_MINOR
+static_assert(false, "ARBORX_VERSION_MINOR macro is not defined!");
+#endif
+
+#ifndef ARBORX_VERSION_PATCH
+static_assert(false, "ARBORX_VERSION_PATCH macro is not defined!");
+#endif
```

Comment:
These are fine if you really want them but it is covered by the static assertions below.